File: procapp-backend/src/main/java/lk/maga/procapp/dto/ApiError.java
package lk.maga.procapp.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.Map;

@Getter
@AllArgsConstructor
public class ApiError {
    private String message;
    private int status;
    private Map<String, String> fieldErrors;
    
}

File: procapp-frontend/src/api/apiError.ts
export class ApiError extends Error {
  status: number
  fieldErrors?: Record<string, string>

  constructor(message: string, status = 400, fieldErrors?: Record<string, string>) {
    super(message)
    this.name = 'ApiError'
    this.status = status
    this.fieldErrors = fieldErrors
  }
}
